app-text/mandoc/mandoc-1.14.3.recipe

```diff
@@ -0,0 +1,113 @@
+SUMMARY="A suite of tools compiling mdoc from the OpenBSD project"
+DESCRIPTION="A suite of tools compiling mdoc, the roff macro language of \
+choice for BSD manual pages, and man, the predominant historical language for \
+UNIX manuals. It is small, ISO C, ISC-licensed, and quite fast. The main \
+component of the toolset is the mandoc utility program, based on the libmandoc \
+validating compiler, to format output for UTF-8 and ASCII UNIX terminals, \
+HTML 5, PostScript, and PDF."
+HOMEPAGE="http://mandoc.bsd.lv/"
+# Copyrights should be updated on every new release
+# http://mandoc.bsd.lv/cgi-bin/cvsweb/LICENSE
+COPYRIGHT="2008-2012, 2014 Kristaps Dzonsons
+	2010-2017 Ingo Schwarze
+	2009, 2010, 2011, 2012 Joerg Sonnenberger
+	2013 Franco Fichtner
+	2014 Baptiste Daroussin
+	2016 Ed Maste
+	2017 Michael Stapelberg
+	1999, 2004 Marc Espie
+	1998, 2004, 2010 Todd C. Miller
+	2008, 2017 Otto Moerbeek
+	2004 Ted Unangst
+	1994 Christos Zoulas
+	2003, 2007, 2008, 2014 Jason McIntyre
+	2014 Baptiste Daroussin
+	1989, 1990, 1993, 1994 The Regents of the University of California
+	1994 Christos Zoulas"
+LICENSE="ISC
+	BSD (2-clause)
+	BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="http://mandoc.bsd.lv/snapshots/mandoc-$portVersion.tar.gz"
+CHECKSUM_SHA256="0b0c8f67958c1569ead4b690680c337984b879dfd2ad4648d96924332fd99528"
+PATCHES="mandoc-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	mandoc$secondaryArchSuffix = $portVersion
+	cmd:apropos = $portVersion
+	cmd:demandoc = $portVersion
+	cmd:makewhatis = $portVersion
+	cmd:man = $portVersion
+	cmd:mandoc = $portVersion
+	cmd:mandocd = $portVersion
+	cmd:soelim = $portVersion
+	cmd:whatis = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+CONFLICTS="
+	cmd:man
```

Comment:
I though pkgman would not correctly handle the `cmd:man` in `CONFLICTS`, but it did :)
(`pkgman` correctly asked me the permission to uninstall `man` when I tried to install `mandoc`, and also correctly asked me the permission to uninstall `mandoc` when I tried to install `man`.)

But I still think it would be better to make the following change:
```diff
CONFLICTS="
-	cmd:man
+	man
 	"
```
This is beacause the `CONFLICTS` we have in **_all_** other recipes **_all_** use package names. So using a `cmd:` item here would mean we would have to be very careful about this in future modifications of pkgman.

Reply:
From [PackageManagement/BuildingPackages][0]

>     entity		::= entity_name [ "=" version_ref ] [ ( "compat" | "compatible" ) ">=" version_ref ]
>     entity_name	::= [ entity_type ":" ] entity_name_char+
>     entity_type	::= "lib" | "cmd" | "app" | "add_on"

>     required_entity		::= entity_name [ version_operator version_ref [ "base" ] ]
>     version_operator	::= "<" | "<=" | "==" | "!=" | ">=" | ">"

> - `conflicts`: A list of entities that this package conflicts with (i.e. only one of both can be installed at any time). The list elements must have the `required_entity` format. 

This means `cmd:man` as `CONFLICTS` will always be supported. And as in `REQUIRES`, we should be specific with what we `CONFLICTS`.

[0]: https://dev.haiku-os.org/wiki/Obsolete/MovedToTree/PackageManagement/BuildingPackages

Reply:
Well, you are right in theory. But I guess the `CONFLICTS` variable was meant to be used with package names only. If it had been meant for `cmd:` items, then we could just use the `cmd:` in `PROVIDES` and not bother to define any `CONFLICTS`.
Moreover, there is a small advantage to using package names: there is no need to do a search.

Since we have no other recipe with `cmd:` items in a `CONFLICTS`, I'd like to get the green lights from one or two key contributors. If nobody objects having a `cmd:man` in mandoc's `CONFLICTS` then I'll be OK with that :)
(But I guess it will be better to switch to the variant.)

Reply:
@pulkomandy & @korli, are you OK with having `cmd:man` in `CONFLICTS` ?
It is working fine and, as I wrote previously, pkgman correctly handles this.

I'm asking the question because this would be the first recipe to use a `cmd:` item in a `CONFLICTS`.
I'm OK with this, but I'd like to have your thoughts before we merge. Thanks!